framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2343,10 +2343,12 @@ public Void visitAnnotation(AnnotationTree tree, Void p) {
      */
     @Override
     public Void visitConditionalExpression(ConditionalExpressionTree tree, Void p) {
-        AnnotatedTypeMirror cond = atypeFactory.getAnnotatedType(tree);
-        this.commonAssignmentCheck(cond, tree.getTrueExpression(), "conditional.type.incompatible");
+        AnnotatedTypeMirror condTrue = atypeFactory.getAnnotatedType(tree);
+        AnnotatedTypeMirror condFalse = atypeFactory.getAnnotatedType(tree);
```

Comment:
Thanks for tracking down this problem and this fix!
I think the root cause of this issue is: https://github.com/eisop/checker-framework/blob/b7745d9fe37fde1f05525c805091d0128fdd804a/checker/src/main/java/org/checkerframework/checker/nullness/NullnessVisitor.java#L247C62-L247C62
where the `commonAssignmentCheck` changes the type of an argument.
I think we should fix that issue - `commonAssignmentCheck` shouldn't change the type. Instead, the NullnessTransfer should properly handle `PolyNull` in the two branches.
Then here we only need one `cond` type, which will be the LUB type of the two branches. This fix works, but it is very confusing why we need to determine the type for the same tree twice. So let's look into properly fixing this.

Reply:
@wmdietl A question here, if we are planning to use the LUB type of the two branches here, I think we are going to have a false positive in this example:
```
    @PolyNull String toLowerCaseA(@PolyNull String text) {
        return text == null ? null : text.toLowerCase();
    }
```
The LUB type would be `@Nullable`. And we check ` text.toLowerCase()`, there would be a dereference.of.nullable error. (At least I do not want this error message).

Reply:
The type of the conditional expression, that is, the type for the whole expression `text == null ? null : text.toLowerCase()`, should be `@PolyNull String`.
The type in the true and false branch can still have refinements. In the then branch we know `text` is null and in the else branch we know `text` is non-null. So no dereference error should be raised.

Reply:
Get it. Thanks.